hwloc_xml_parser/topology.py

```diff
@@ -173,3 +173,24 @@ def recurse_pus(self) -> typing.Generator:
         for core in self.recurse_cores():
             for pu in core.pus:
                 yield pu
+
+    @typeguard.typechecked
+    def get_num_packages(self) -> int:
+        """
+        Returns the number of packages.
+        """
+        return len(self.packages)
+
+    @typeguard.typechecked
+    def get_num_cores_per_package(self, *, logical_index : int = 0) -> int:
+        """
+        Returns the number of cores per package.
```

Comment:
```suggestion
        Returns the number of cores for package `logical_index`.
```
(or something like this)

Reply:
I changed it a bit. It is now the `Package` object itself that has a `get_num_cores` function. The `Machine` object has such a function too, and this one aggregates over all the packages. :bar_chart:

Reply:
OK :checkered_flag: